src/main/cpp/checkpoint/CheckPoint.cpp

```diff
@@ -984,10 +1106,381 @@ boost::gregorian::date CheckPoint::GetLastDate()
+	H5Awrite(attr, H5T_NATIVE_UINT, uints);
+	H5Sclose(dataspace);
+	H5Aclose(attr);
+
+	dims = 3;
+	dataspace = H5Screate_simple(1, &dims, nullptr);
+	attr = H5Acreate2(group, "doubles", H5T_NATIVE_DOUBLE, dataspace, H5P_DEFAULT, H5P_DEFAULT);
+	H5Awrite(attr, H5T_NATIVE_DOUBLE, doubles);
+	H5Sclose(dataspace);
+	H5Aclose(attr);
+
+	dims = prefix.size();
+	dataspace = H5Screate_simple(1, &dims, nullptr);
+	attr = H5Acreate2(group, "prefix", H5T_NATIVE_CHAR, dataspace, H5P_DEFAULT, H5P_DEFAULT);
+	H5Awrite(attr, H5T_NATIVE_CHAR, prefix.c_str());
+	H5Sclose(dataspace);
+	H5Aclose(attr);
+}
+
+void CheckPoint::LoadFile(const std::string& filestr, const std::string& setname)
+{
+	htri_t exist = H5Lexists(m_file, setname.c_str(), H5P_DEFAULT);
+	if (exist <= 0) {
+		FATAL_ERROR("Invalid file");
+	}
+	boost::filesystem::path filename(filestr);
+	/*
+	Piece of code based on code on
+	https://lists.hdfgroup.org/pipermail/hdf-forum_lists.hdfgroup.org/2010-June/003208.html
+	*/
+	std::vector<char> data;
+	hid_t did;
+	herr_t err = 0;
+	hid_t spaceId;
+	hid_t dataType = H5T_NATIVE_CHAR;
+
+	// std::cout << "HDF5 Data Type: " <<	H5Lite::HDFTypeForPrimitiveAsStr(test) << std::endl;
+	/* Open the dataset. */
+	// std::cout << "  Opening " << dsetName << " for data Retrieval.  "<< std::endl;
+	did = H5Dopen(m_file, setname.c_str(), H5P_DEFAULT);
+
+	if (did < 0) {
+		std::cout << " Error opening Dataset: " << did << std::endl;
+		return;
+	}
+
+	std::string extension;
+	hid_t attr = H5Aopen(did, "extension", H5P_DEFAULT);
+
+	std::unique_ptr<H5A_info_t> info = std::make_unique<H5A_info_t>();
+
+	H5Aget_info(attr, info.get());
+	std::vector<char> extensionVector(info->data_size, '\0');
+	H5Aread(attr, H5T_NATIVE_CHAR, extensionVector.data());
+	H5Aclose(attr);
+
+	extension = std::string(extensionVector.begin(), extensionVector.end());
+
+	filename.replace_extension(extension);
+
+	std::ofstream out(filename.string(), std::ios::out | std::ios::trunc);
+	if (did >= 0) {
+		spaceId = H5Dget_space(did);
+		if (spaceId > 0) {
+			int32_t rank = H5Sget_simple_extent_ndims(spaceId);
+			if (rank > 0) {
+				std::vector<hsize_t> dims;
+				dims.resize(rank); // Allocate enough room for the dims
+				rank = H5Sget_simple_extent_dims(spaceId, &(dims.front()), NULL);
+				hsize_t numElements = 1;
+				for (std::vector<hsize_t>::iterator iter = dims.begin(); iter < dims.end(); ++iter) {
+					numElements = numElements * (*iter);
+				}
+				// std::cout << "NumElements: " << numElements << std::endl;
+				// Resize the vector
+				data.resize(static_cast<int>(numElements));
+				// for (uint32_t i = 0; i<numElements; ++i) { data[i] = 55555555;}
+				err = H5Dread(did, dataType, H5S_ALL, H5S_ALL, H5P_DEFAULT, &(data.front()));
+				if (err < 0) {
+					std::cout << "Error Reading Data." << std::endl;
+				}
+			}
+			err = H5Sclose(spaceId);
+			if (err < 0) {
+				std::cout << "Error Closing Data Space" << std::endl;
```

Comment:
Please use `std::cerr` to report errors and/or diagnostics. `std::cout` is intended for actual application output.